public/modules/custom/grants_profile/src/Form/GrantsProfileFormUnregisteredCommunity.php

```diff
@@ -0,0 +1,1259 @@
+
+  /**
+   * Access to grants profile services.
+   *
+   * @var \Drupal\grants_profile\GrantsProfileService
+   */
+  protected GrantsProfileService $grantsProfileService;
+
+  /**
+   * Constructs a new GrantsProfileForm object.
+   *
+   * @param \Drupal\Core\TypedData\TypedDataManager $typed_data_manager
+   *   Data manager.
+   * @param \Drupal\grants_profile\GrantsProfileService $grantsProfileService
+   *   Grants profile service.
+   */
+  public function __construct(TypedDataManager $typed_data_manager, GrantsProfileService $grantsProfileService) {
+    $this->typedDataManager = $typed_data_manager;
+    $this->grantsProfileService = $grantsProfileService;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container): GrantsProfileFormPrivatePerson|static {
+    return new static(
+      $container->get('typed_data_manager'),
+      $container->get('grants_profile.service')
+    );
+  }
+
+  /**
+   * Helper method so we can have consistent dialog options.
+   *
+   * @return string[]
+   *   An array of jQuery UI elements to pass on to our dialog form.
+   */
+  public static function getDataDialogOptions(): array {
+    return [
+      'width' => '33%',
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'grants_profile_unregistered_community';
+  }
+
+  /**
+   * {@inheritdoc}
+   *
+   * @throws \GuzzleHttp\Exception\GuzzleException
+   */
+  public function buildForm(array $form, FormStateInterface $form_state): array {
+
+    $selectedRoleData = $this->grantsProfileService->getSelectedRoleData();
+
+    // Load grants profile.
+    $grantsProfile = $this->grantsProfileService->getGrantsProfile($selectedRoleData, TRUE);
+
+    // If no profile exist.
+    if ($grantsProfile == NULL) {
+      // Create one and.
+      $grantsProfile = $this->grantsProfileService->createNewProfile($selectedRoleData);
+    }
+
+    if ($grantsProfile == NULL) {
+      return [];
+    }
+
+    // Get content from document.
+    $grantsProfileContent = $grantsProfile->getContent();
+
+    $storage = $form_state->getStorage();
+    $storage['profileDocument'] = $grantsProfile;
+
+    // Use custom theme hook.
+    $form['#theme'] = 'own_profile_form';
+    $form['#tree'] = TRUE;
+
+    $form['#after_build'] = ['Drupal\grants_profile\Form\GrantsProfileFormUnregisteredCommunity::afterBuild'];
+
+    $form['companyNameWrapper'] = [
+      '#type' => 'webform_section',
+      '#title' => $this->t('Community name'),
+    ];
+    $form['companyNameWrapper']['companyName'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Community name'),
+      '#default_value' => $grantsProfileContent['companyName'],
+    ];
+
+    $form['newItem'] = [
+      '#type' => 'hidden',
+      '#value' => NULL,
+    ];
+    $newItem = $form_state->getValue('newItem');
+
+    $this->addAddressBits($form, $form_state, $grantsProfileContent['addresses'], $newItem);
```

Comment:
Muokkauslomakkeella tulee nämä kentät aivan oikein mutta kun rekisteröimättömän yrityksen tietoja katselee niin silloin tulee näkyviin kenttiä joita ei voi muokata.
<img width="827" alt="Screenshot 2023-04-20 at 16 35 52" src="https://user-images.githubusercontent.com/6604896/233383193-641913a4-815f-4f9c-8625-454bcd5b9417.png">

Reply:
https://github.com/City-of-Helsinki/hel-fi-drupal-grants/pull/334